dali/python/nvidia/dali/pipeline.py

```diff
@@ -990,3 +996,62 @@ def iter_setup(self):
         For example, one can use this function to feed the input
         data from NumPy arrays."""
         pass
+
+
+def _discriminate_args(**kwargs):
+    """Split args on those applicable to Pipeline constructor and the rest."""
+    fca = inspect.getfullargspec(Pipeline.__init__)  # Full Ctor Args
+    ctor_args = dict(filter(lambda x: x[0] in fca.args or x[0] in fca.kwonlyargs, kwargs.items()))
+    fn_args = dict(filter(lambda x: x[0] not in ctor_args.keys(), kwargs.items()))
+    return ctor_args, fn_args
+
+
+def pipeline(fn=None, **pipeline_kwargs):
+    """
+    Decorator for wrapping functions that define DALI pipelines.
```

Comment:
Maybe add something like: `Decorated functions becomes a pipeline factory`? Or `Decorated function will return instances of Pipeline with processing defined by the function`.

Reply:
The latter - or sth like:
```suggestion
    A decorator which creates a factory of DALI pipelines whose processing graph is defined by the decorated function.
```

Reply:
Done